yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -819,12 +819,22 @@ private[spark] class Client(
       javaOpts += "-XX:CMSIncrementalDutyCycle=10"
     }
 
-    // Include driver-specific java options if we are launching a driver
-    if (isClusterMode) {
```

Comment:
We have introduced a generic check in the SparkSubmitCommandBuilder and SparkClassCommandBuilder which checks for Xmx settings being passed by the user before launching the driver. Earlier changes in Client.scala have been restored to the previous behavior where we are not making an explicit check based on the cluster or client mode.